### PR TITLE
feat!: support-deal-whitlists [CHAIN-368]

### PR DESCRIPTION
**Pull request by AlcibiadesCleinias**
# BreakingChange (new contract)
- fix CHAIN-400 (bug in Matcher.matchDeal when match with whitelisted deal)

# ChangeLog
- fix blacklist in matcher
- add whitelist direct check (no need in CC Active status in the case of Deal with whitelisted Providers)
- add docstrings

# ToDo
- [x] redeployed contracts on stage and dar 
- [x] optional: add integrational test for this case of match

### CODE REVIEW

File: ts-client/tests/integration/fixture.ts [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+            paymentTokenAddress: paymentToken,
+            minProtocolVersion: 1,
+            maxProtocolVersion: 1,
+        },
+        providerToBeMatched: {
+            providerAddress: "",
+            peerIds: [ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32))],
+            computeUnitsPerPeers: [
+                [ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32))],
+                [ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32)), ethers.hexlify(ethers.randomBytes(32))],
```
Comment (akim-bow) [inline ts-client/tests/integration/fixture.ts:84]: ditto

File: ts-client/tests/integration/fixture.ts [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+        },
+        dealToMatchWithWhiteListedProvider: {
+            developerAddress: "",
+            appCID: { prefix: ethers.hexlify(ethers.randomBytes(4)), hash: ethers.hexlify(ethers.randomBytes(32)), description: "dealAPPCID"},
+            paymentTokenAddress: paymentToken,
+            // pricePerWorkerEpoch * newTargetWorkers * core.minDealDepositedEpoches();
+            depositAmount: null,
+            minWorkers: 2,
+            targetWorkers: 2,
+            maxWorkerPerProvider: 2,
```
Comment (akim-bow) [inline ts-client/tests/integration/fixture.ts:113]: move to constant?

Comment (AlcibiadesCleinias) [reply ts-client/tests/integration/fixture.ts:113]: no, for now it is better to refer to this fixture initialisation to check all params, also I do not relay exactly on 2, I relay on minWorkers and etc, MB in the future

**Review by akim-bow — APPROVED** [review]

File: ts-client/tests/integration/fixture.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+                new Array(2).fill("0").map(() => {return ethers.hexlify(ethers.randomBytes(32))}),
+                new Array(2).fill("0").map(() => {return ethers.hexlify(ethers.randomBytes(32))}),
+            ],
+            effectors,
+            minPricePerEpoch: minPricePerEpoch.toString(),
+            paymentTokenAddress: paymentToken,
+            minProtocolVersion: 1,
+            maxProtocolVersion: 1,
+        },
+        dealToMatchWithWhiteListedProvider: {
```
Comment (folex) [inline ts-client/tests/integration/fixture.ts:101]: I don't see whitelisted provider here

**Pull request merged**